src/imitation/data/rollout.py

```diff
@@ -335,6 +340,7 @@ def generate_trajectories(
         may be collected to avoid biasing process towards short episodes; the user
         should truncate if required.
     """
+    rng = rng or np.random.RandomState()
```

Comment:
As I discussed before with MCE IRL, this is a semantic change that means we no longer benefit from global seeding. In TabularPolicy it did not matter so much, that code path was barely even used. But rollouts are a pretty core part of it, and this change will make our experiments less replicalbe (previously setting the global Sacred seed to the same value should be largely enough.)

Main options seem to either be to just ignore the type error (e.g. add ugly cast), as `np.random` does in fact implement all the methods of `RandomState` that we need, or change callsites to `rollout` to take in an explicit `rng` (this is the cleaner and more replicable thing long-term, since global sources of randomness are easily polluted anyway).

Reply:
I am inclined to go for the second.

Reply:
To be fair, I don't see how relying on the global seed makes anything more replicable. The global seed is drawn from /dev/random just like the local seed when creating an instance of `RandomState` AFAIK. The only benefit is that you can fix the global seed when you start the program, but I don't see that we do that anywhere except for one test.

Reply:
After thinking about this, I opted for the design decision to force all functions to require explicit random states everywhere. Then the user can decide on the high-level API whether to use a fixed seed or to take it from /dev/random every time the script is run. This explicit consent ensures that users know what parts of their code are using random number generators, and whether this behavior is being controlled or not. It also allows us to explicitly choose for each test whether we want a fixed or a random seed, through a fixture that I have added. 

Let me know if you disagree with this, we can have a more detailed discussion. I know it's a fairly big change tangential to the PR, but in a sense ensuring type safety is strongly related to ensuring that default arguments are not passed tacitly when they shouldn't be, and seeds are a very good case of that IMO.